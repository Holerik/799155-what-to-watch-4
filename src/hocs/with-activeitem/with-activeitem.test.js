// with-activeitem.test.js
import React from 'react';
import renderer from 'react-test-renderer';
import withActiveItem from './with-activeitem.jsx';
import PropTypes from 'prop-types';

const genresList = [`All genres`, `Drama`, `Crime`];

const MockComponent = (props) => {
  return (
    <ul>
      {props.listItems.map((item, index) => {
        return (
          <React.Fragment key={index}>
            <li>
              <a>{item}</a>
            </li>
          </React.Fragment>
        );
      })}
    </ul>
  );
};

MockComponent.propTypes = {
  listItems: PropTypes.arrayOf(PropTypes.string),
  currentActiveItem: PropTypes.number,
  setActiveItem: PropTypes.func,
  maxItemsCount: PropTypes.number,
};

const MockComponentWrapped = withActiveItem(MockComponent);

describe(`withActiveItem tests`, () => {
  it(`Should correctly render genres list`, () => {
    const tree = renderer.create((
      <MockComponentWrapped
        listItems={genresList}
        currentActiveItem={0}
        setActiveItem={() => {}}
        maxItemsCount={3}
        id={0}
      />), {
      createNodeMock() {
        return {};
      }
    }).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
