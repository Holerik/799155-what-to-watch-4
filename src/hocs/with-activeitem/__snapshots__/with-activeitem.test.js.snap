// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`withActiveItem tests Should correctly render genres list 1`] = `
<ul>
  <li>
    <a>
      All genres
    </a>
  </li>
  <li>
    <a>
      Drama
    </a>
  </li>
  <li>
    <a>
      Crime
    </a>
  </li>
</ul>
`;
